saq/queue/postgres.py

```diff
@@ -0,0 +1,334 @@
+        logger.info("Finished %s", job.info(logger.isEnabledFor(logging.DEBUG)))
+
+    async def dequeue(self, timeout: float = 0) -> Job | None:
+        async with self.pool.connection() as conn, conn.cursor() as cursor:
+            await cursor.execute(
+                f"""
+                UPDATE {JOBS_TABLE} SET status = %(active)s
+                WHERE key = (
+                  SELECT key FROM {JOBS_TABLE}
+                  WHERE status = %(queued)s AND queue = %(queue)s and EXTRACT(EPOCH FROM now()) >= scheduled
+                  FOR UPDATE SKIP LOCKED
+                  LIMIT 1
+                )
+                RETURNING job
+                """,
+                {"active": Status.ACTIVE, "queued": Status.QUEUED, "queue": self.name},
+            )
+            job = await cursor.fetchone()
+            if job:
+                return self.deserialize(job[0])
+        return None
+
+    async def enqueue(self, job_or_func: str | Job, **kwargs: t.Any) -> Job | None:
+        job = self._create_job_for_enqueue(job_or_func, **kwargs)
+
+        await self._before_enqueue(job)
+
+        async with self.pool.connection() as conn, conn.cursor() as cursor:
+            await cursor.execute(
+                f"""
+                INSERT INTO {JOBS_TABLE} (key, function, job, queue, status, scheduled)
+                VALUES (%(key)s, %(function)s, %(job)s, %(queue)s, %(status)s, %(scheduled)s)
+                ON CONFLICT (key) DO UPDATE
+                SET function = %(function)s, job = %(job)s, queue = %(queue)s, status = %(status)s, scheduled = %(scheduled)s
+                """,
+                {
+                    "key": job.key,
+                    "function": job.function,
+                    "job": Jsonb(self.serialize(job)),
+                    "queue": self.name,
+                    "status": job.status,
+                    "scheduled": job.scheduled,
+                },
+            )
+
+        logger.info("Enqueuing %s", job.info(logger.isEnabledFor(logging.DEBUG)))
+        return job
+
+    async def get_many(self, keys: Iterable[str]) -> list[bytes | None]:
+        async with self.pool.connection() as conn, conn.cursor() as cursor:
+            await cursor.execute(
+                f"""
+                SELECT job FROM {JOBS_TABLE}
+                WHERE key in %(keys)s
+                """,
+                {"keys": keys},
+            )
+            return [row[0] for row in await cursor.fetchmany()]
+
+    async def delete(self, key: str) -> None:
+        async with self.pool.connection() as conn, conn.cursor() as cursor:
+            await cursor.execute(
+                f"""
+                DELETE FROM {JOBS_TABLE}
+                WHERE key = %(key)s
+                """,
+                {"key": key},
+            )
+
+    def _escape_channel_name(self, key: str) -> str:
+        return f"channel_{key.replace('-', '_')}"
+
+    def _unescape_channel_name(self, channel: str) -> str:
```

Comment:
why do you need this?

Reply:
Postgres channel names have the same restrictions as table names. This was a hack to work around it. I'll add a better way to encode channel names.

Reply:
Realized I could just quote the channel name and not worry about this...